dashboard/lib/services/lesson_plan_pdfs.rb

```diff
@@ -135,18 +136,43 @@ def self.get_pathname(lesson)
       return Pathname.new(File.join(lesson.script.name, version_number, filename))
     end
 
+    # Build the full path of the student lesson plan PDF for the given lesson. This
+    # will be based on not only the lesson's script but also the current
+    # version of the script in the environment.
+    #
+    # Expect this to look something like
+    # <Pathname:csp1-2021/20210216001309/Welcome to CSP-Student.pdf>
+    def self.get_student_pathname(lesson)
+      return nil unless lesson&.script&.seeded_from
+      version_number = Time.parse(lesson.script.seeded_from).to_s(:number)
+      filename = ActiveStorage::Filename.new(lesson.key + "-Student" + ".pdf").sanitized
```

Comment:
This naming scheme runs the risk of collision, and is just generally not as closely based on the database-enforced unique identifiers as I'd like it to be. I'd much prefer something like `csp1-2021/20210216001309/student/Welcome to CSP.pdf`